cmd/agent/app/options/options.go

```diff
@@ -0,0 +1,186 @@
+package options
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/spf13/pflag"
+	"google.golang.org/grpc"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/apiserver-network-proxy/pkg/agent"
+	"sigs.k8s.io/apiserver-network-proxy/pkg/util"
+)
+
+type GrpcProxyAgentOptions struct {
+	// Configuration for authenticating with the proxy-server
+	AgentCert string
+	AgentKey  string
+	CaCert    string
+
+	// Configuration for connecting to the proxy-server
+	ProxyServerHost string
+	ProxyServerPort int
+	AlpnProtos      []string
+
+	// Ports for the health and admin server
+	HealthServerPort int
+	AdminServerPort  int
+	// Enables pprof at host:adminPort/debug/pprof.
+	EnableProfiling bool
+	// If EnableProfiling is true, this enables the lock contention
+	// profiling at host:adminPort/debug/pprof/block.
+	EnableContentionProfiling bool
+
+	AgentID          string
+	AgentIdentifiers string
+	SyncInterval     time.Duration
+	ProbeInterval    time.Duration
+	SyncIntervalCap  time.Duration
+
+	// file contains service account authorization token for enabling proxy-server token based authorization
+	ServiceAccountTokenPath string
+}
+
+func (o *GrpcProxyAgentOptions) ClientSetConfig(dialOptions ...grpc.DialOption) *agent.ClientSetConfig {
+	return &agent.ClientSetConfig{
+		Address:                 fmt.Sprintf("%s:%d", o.ProxyServerHost, o.ProxyServerPort),
+		AgentID:                 o.AgentID,
+		AgentIdentifiers:        o.AgentIdentifiers,
+		SyncInterval:            o.SyncInterval,
+		ProbeInterval:           o.ProbeInterval,
+		SyncIntervalCap:         o.SyncIntervalCap,
+		DialOptions:             dialOptions,
+		ServiceAccountTokenPath: o.ServiceAccountTokenPath,
+	}
+}
+
+func (o *GrpcProxyAgentOptions) Flags() *pflag.FlagSet {
+	flags := pflag.NewFlagSet("proxy-agent", pflag.ContinueOnError)
+	flags.StringVar(&o.AgentCert, "agent-cert", o.AgentCert, "If non-empty secure communication with this cert.")
```

Comment:
I realize this is largely a forklift but slightly nervous about making sure all the flags are maintained (i.e. binaries are backward compatible). Any chance of tests for the Flags functions?

Reply:
I checked that I didn't modify the flags' names. And I tested the flags locally.